tests/fixtures/datacite_fixtures.py

```diff
@@ -0,0 +1,262 @@
+    """
+    return {
+        "id": "10.7303/syn12345",
+        "type": "dois",
+        "attributes": {
+            **base_doi_attributes,
+            "doi": "10.7303/syn12345",
+            "prefix": "10.7303",
+            "suffix": "syn12345",
+            "creators": [
+                {
+                    "name": "Test Author",
+                    "nameType": "Personal",
+                    "givenName": "Test",
+                    "familyName": "Author",
+                    "affiliation": [],
+                    "nameIdentifiers": []
+                }
+            ],
+            "titles": [{"title": "Test Dataset"}],
+            "publisher": "Synapse",
+            "publicationYear": 2024,
+            "types": {
+                "ris": "GEN",
+                "bibtex": "misc",
+                "citeproc": "article",
+                "schemaOrg": "Dataset",
+                "resourceTypeGeneral": "Dataset"
+            },
+            "url": "https://repo-prod.prod.sagebase.org/repo/v1/doi/locate?id=syn12345&type=ENTITY",
+            "state": "findable",
+            "created": "2024-01-15T10:30:00Z",
+            "registered": "2024-01-15T10:30:00Z",
+            "published": None,
+            "updated": "2024-01-15T10:30:00Z"
+        },
+        "relationships": base_doi_relationships
+    }
+
+
+def _create_minimal_doi(doi_id: str, suffix: str, title: str, author: str) -> Dict[str, Any]:
+    """Helper function to create minimal DOI object for testing.
+    
+    Args:
+        doi_id: Full DOI identifier.
+        suffix: DOI suffix (after prefix).
+        title: Dataset title.
+        author: Creator name.
+        
+    Returns:
+        Minimal DOI object with essential fields for testing.
+    """
+    return {
+        "id": doi_id,
+        "type": "dois",
+        "attributes": {
+            "doi": doi_id,
+            "prefix": "10.7303",
+            "suffix": suffix,
+            "creators": [{"name": author, "nameType": "Personal"}],
+            "titles": [{"title": title}],
+            "publisher": "Synapse",
+            "publicationYear": 2024,
+            "types": {"resourceTypeGeneral": "Dataset"},
+            "state": "findable",
+            "created": "2024-01-15T10:30:00Z",
+            "updated": "2024-01-15T10:30:00Z"
+        }
+    }
+
+
+@pytest.fixture
+def sample_doi_objects() -> List[Dict[str, Any]]:
+    """Multiple DOI objects for pagination testing.
+    
+    Returns:
+        List of DOI objects with varying IDs.
+    """
+    return [
+        _create_minimal_doi(
+            doi_id=f"10.7303/syn{i:05d}",
+            suffix=f"syn{i:05d}",
+            title=f"Dataset {i}",
+            author=f"Author {i}"
+        )
+        for i in range(1, 11)
+    ]
+
+
+@pytest.fixture
+def mock_api_response_full_page(sample_doi_objects) -> Dict[str, Any]:
+    """Full page API response (page size matches data length).
```

Comment:
Should we have a test case where the data length for the full page is > the page size queried?

Reply:
Resolved by previous related comment